### PR TITLE
fix tests after migrate aggregations on references of columns and tables

### PR DESCRIPTION
**Pull request by dbulahov**
Signed-off-by: dbulahov <bulahovdenis@gmail.com>

### CODE REVIEW

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRecognizer.java [hunk 1]
```diff
@@ -495,7 +495,7 @@ public int compare(
         Optional<Map.Entry<org.eclipse.daanse.rdb.structure.api.model.Column, org.eclipse.daanse.rdb.structure.api.model.Column>> op = measuresFactCount.entrySet()
         		.stream().filter(e -> e.getKey().getName().equals(measureName)).findAny();         
         String factCountColumnName;
-        if (op.isPresent() && Util.isEmpty(op.get().getValue().getName())) {
+        if (op.isPresent() && !Util.isEmpty(op.get().getValue().getName())) {
```
Comment (stbischof) [inline mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRecognizer.java:498]: How was this in history? Sure about that?

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java [hunk 2]
```diff
@@ -1209,7 +1210,7 @@ protected Recognizer.Matcher getMeasureFactCountMatcher() {
                 public boolean matches(String name) {
                     HashSet<Column> measuresFactCountSet =
                             new HashSet<>(measuresFactCount.values());
-                    return measuresFactCountSet.contains(name);
+                    return measuresFactCountSet.stream().anyMatch(c -> name.equals(c.getName()));
```
Comment (stbischof) [inline mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java:1213]: Maybe fist use
.map(Column::getName).anyMatch


I not sure please test

Comment (stbischof) [reply mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java:1213]: And also may .filter( objects:notnull)

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java [hunk 3]
```diff
@@ -1325,7 +1330,7 @@ public void validate(final MessageRecorder msgRecorder) {
                     // Is the level foreign key name a duplicate
                     if (columnsToObjects.containsKey(level.getColumnName().getName())) {
                         Level l = (Level)
-                            columnsToObjects.get(level.getColumnName());
+                            columnsToObjects.get(level.getColumnName().getName());
```
Comment (stbischof) [inline mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java:1333]: Rename this.

getColumn().getname()

Pleae verif all uses

**Review by stbischof — COMMENTED** [review]
See comments

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java (hunk 3 above)
Comment (dbulahov) [reply mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java:1333]: agree

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRecognizer.java (hunk 1 above)
Comment (dbulahov) [reply mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRecognizer.java:498]: `@Override
	protected String getFactCountColumnName
            (final JdbcSchema.Table.Column.Usage aggUsage) {
        String measureName = aggUsage.getColumn().getName();
        Map<String, String> measuresFactCount = tableDef.getMeasuresFactCount();
        String factCountColumnName = measuresFactCount.get(measureName);
        if (Util.isEmpty(factCountColumnName)) {
            factCountColumnName = getFactCountColumnName();
        }
        return factCountColumnName;
    }`

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java (hunk 2 above)
Comment (dbulahov) [reply mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java:1213]: agree

Comment (dbulahov) [reply mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java]: done

File: mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java (hunk 3 above)
Comment (dbulahov) [reply mondrian/src/main/java/mondrian/rolap/aggmatcher/ExplicitRules.java]: done

**Pull request merged**